### ISSUE TITLE
Introduce PlanCompiler that compiles Dolphin-level plan to ET-level plan

### ISSUE THREAD

**Issue by wynot12** [snuspl/cruise#1097]
Dolphin has a plan implementation, which is a collection of ADD/DEL/MOVE for each namespace (e.g., Server and Worker).

ET has following following operations, which is lower-level than Dolphin's.
- Executor (ALLOCATE/DEALLOCATE)
- Table (MOVE/SUBSCRIBE/UNSUBSCRIBE)
- Task (Start/Stop)

Differently from Dolphin, ET does not have a notion of namespace. In addition, ET does not restrict one executor to have only one table and task is not tied with executor.

In other words, ET-level plan package is not dolphin-specific. So Dolphin needs to provide a component that translate Dolphin-level plan to ET-level plan.
